packages/views.api.unstable/src/manifest.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  EventDefinition,
+  ParameterConfig,
+  ParameterDefinition,
+} from "./config.js";
+
+export interface ViewManifest {
+  views: Record<string, ViewManifestConfig>;
+}
+
+export interface ViewManifestConfig {
+  rid: string;
+  version: string;
+  entrypointJs: string[];
+  entrypointCss?: string[];
+  parameters: Record<string, ParameterDefinition>;
+  events: Record<string, EventDefinition<ParameterConfig>>;
```

Comment:
Are any of these (looking at parameters/events) specific to workshop or will all host applications be able to provide the view with these? I'm wondering if we need to declare the type of view if there are view type specific configs and then there will be validation on publish that the type matches up with the view rid it publishes to

Reply:
I would expect any host application to need to support passing in parameters/accepting events, not just Workshop. I think it's ok for that translation to happen internally for now while we just have 1